kernel/Base/PhysicalFace.h

```diff
@@ -156,27 +162,22 @@ class PhysicalFace final {
 
     /// curl of basis function i at the current reference point; indexing
     /// functions in the right element after functions in the left element
-    const LinearAlgebra::SmallVector<DIM>& basisFunctionCurl(std::size_t i);
     const LinearAlgebra::SmallVector<DIM>& basisFunctionCurl(
-        std::size_t i, std::size_t unknown);
+        std::size_t i, std::size_t unknown = 0);
 
     /// curl of basis function i at the current reference point; indexing the
     /// left and the right element separately
-    const LinearAlgebra::SmallVector<DIM>& basisFunctionCurl(Side side,
-                                                             std::size_t i);
     const LinearAlgebra::SmallVector<DIM>& basisFunctionCurl(
-        Side side, std::size_t i, std::size_t unknown);
+        Side side, std::size_t i, std::size_t unknown = 0);
 
     /// divergence of basis function i at the current reference point; indexing
     /// functions in the right element after functions in the left element
-    const double& basisFunctionDiv(std::size_t i);
-    const double& basisFunctionDiv(std::size_t i, std::size_t unknown);
+    const double& basisFunctionDiv(std::size_t i, std::size_t unknown = 0);
```

Comment:
I am not a fan of returning const double& instead return just double, but it does not matter anyway. Why the default argument for unknown?

Reply:
The original hpgem started with no unknowns, then transitioned into multiple unknowns but all with the same basis functions to be used for the approximation. So that is where the `basisFunctionDiv(std::size_t i)` signature came from.

The signature with unknown was added when unknowns with different basis functions were introduced. The old function remained which defaulted to using the 0-th unknown, but with a check that the basis functions for all unknowns are the same.

By having the default unknown this keeps compatibility with old codes/the common use case, but reduces the code duplication. The only effective change is that there is no check anymore that all basis functions are the same.